src/include/distributed/query_utils.h

```diff
@@ -0,0 +1,40 @@
+/*-------------------------------------------------------------------------
+ *
+ * query_utils.h
+ *
+ * Declarations for query-walker utility functions and related types.
+ *
+ * Copyright (c) 2019, Citus Data, Inc.
+ *
+ ***-------------------------------------------------------------------------
+ */
+#ifndef QUERY_UTILS_H
+#define QUERY_UTILS_H
+
+#include "postgres.h"
+#include "nodes/pg_list.h"
+
+/* Enum to define execution flow of ExtractRangeTableList */
+typedef enum ExtractMode
+{
+	EXT_RELATION_ENTRIES,
+	EXT_INDEX_ENTRIES,
+	EXT_ALL_ENTRIES
+} ExtractMode;
+
+/* Struct to pass rtable list and execution flow flag to query_walker_tree */
+typedef struct ExtractParams
+{
+	List **rangeTableList;
+	ExtractMode walkerMode;
+} ExtractParams;
```

Comment:
This type of struct is often referred to as a (walker) context in Postgres.

e.g. `ExtractRangeTableWalkerContext`